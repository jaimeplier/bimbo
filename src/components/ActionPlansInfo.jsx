import React, { Component } from 'react';
import Poly from './../utils/i18n';

export default class ActionPlansInfo extends Component {
  render() {
    const { metrics } = this.props;
    return (
      <div className="action-plans-info">
        <div className="title">
          <p>{ Poly.t('Action Plans') }</p>
        </div>
        <div className="main-content">
          <div className="action-plan">
            <p>{ Poly.t('Created') }</p>
            <p className="f-right">{ metrics && metrics.get('created') }</p>
          </div>
          <div className="action-plan border-line">
            <p>{ Poly.t('Completed') }</p>
            <p className="f-right">{ metrics && metrics.get('completed') }</p>
          </div>
          <div className="action-plan">
            <p>{ Poly.t('Pending') }</p>
            <p className="f-right">{ metrics && metrics.get('pending') }</p>
          </div>
        </div>
      </div>
    );
  }
}
