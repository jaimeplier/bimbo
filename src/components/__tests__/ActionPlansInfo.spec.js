import React from 'react';
import { shallow } from 'enzyme';
import ActionPlansInfo from '../ActionPlansInfo';
import { fromJS } from 'immutable';

describe('ActionPlansInfo', () => {

  test('Component should render', () => {
    const actionPlansInfo = shallow(<ActionPlansInfo />);
    expect(actionPlansInfo).toMatchSnapshot();
  })

  test('Component should render metrics', () => {
    const actionPlansInfo = shallow(
      <ActionPlansInfo metrics={fromJS({created: 10, completed: 10, pending: 10})} />
    )
    expect(actionPlansInfo).toMatchSnapshot();
  })

})


