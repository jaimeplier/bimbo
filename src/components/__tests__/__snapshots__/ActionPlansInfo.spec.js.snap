exports[`ActionPlansInfo Component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActionPlansInfo />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title">
          <p>
                    Action Plans
          </p>
</div>,
        <div
          className="main-content">
          <div
                    className="action-plan">
                    <p>
                              Created
                    </p>
                    <p
                              className="f-right" />
          </div>
          <div
                    className="action-plan border-line">
                    <p>
                              Completed
                    </p>
                    <p
                              className="f-right" />
          </div>
          <div
                    className="action-plan">
                    <p>
                              Pending
                    </p>
                    <p
                              className="f-right" />
          </div>
</div>,
      ],
      "className": "action-plans-info",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Action Plans
</p>,
          "className": "title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Action Plans",
          },
          "ref": null,
          "rendered": "Action Plans",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="action-plan">
              <p>
                            Created
              </p>
              <p
                            className="f-right" />
</div>,
            <div
              className="action-plan border-line">
              <p>
                            Completed
              </p>
              <p
                            className="f-right" />
</div>,
            <div
              className="action-plan">
              <p>
                            Pending
              </p>
              <p
                            className="f-right" />
</div>,
          ],
          "className": "main-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Created
</p>,
                <p
                  className="f-right" />,
              ],
              "className": "action-plan",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Created",
                },
                "ref": null,
                "rendered": "Created",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "f-right",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Completed
</p>,
                <p
                  className="f-right" />,
              ],
              "className": "action-plan border-line",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Completed",
                },
                "ref": null,
                "rendered": "Completed",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "f-right",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Pending
</p>,
                <p
                  className="f-right" />,
              ],
              "className": "action-plan",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Pending",
                },
                "ref": null,
                "rendered": "Pending",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "f-right",
                },
                "ref": null,
                "rendered": null,
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title">
            <p>
                        Action Plans
            </p>
</div>,
          <div
            className="main-content">
            <div
                        className="action-plan">
                        <p>
                                    Created
                        </p>
                        <p
                                    className="f-right" />
            </div>
            <div
                        className="action-plan border-line">
                        <p>
                                    Completed
                        </p>
                        <p
                                    className="f-right" />
            </div>
            <div
                        className="action-plan">
                        <p>
                                    Pending
                        </p>
                        <p
                                    className="f-right" />
            </div>
</div>,
        ],
        "className": "action-plans-info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              Action Plans
</p>,
            "className": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Action Plans",
            },
            "ref": null,
            "rendered": "Action Plans",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="action-plan">
                <p>
                                Created
                </p>
                <p
                                className="f-right" />
</div>,
              <div
                className="action-plan border-line">
                <p>
                                Completed
                </p>
                <p
                                className="f-right" />
</div>,
              <div
                className="action-plan">
                <p>
                                Pending
                </p>
                <p
                                className="f-right" />
</div>,
            ],
            "className": "main-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Created
</p>,
                  <p
                    className="f-right" />,
                ],
                "className": "action-plan",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Created",
                  },
                  "ref": null,
                  "rendered": "Created",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "f-right",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Completed
</p>,
                  <p
                    className="f-right" />,
                ],
                "className": "action-plan border-line",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Completed",
                  },
                  "ref": null,
                  "rendered": "Completed",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "f-right",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Pending
</p>,
                  <p
                    className="f-right" />,
                ],
                "className": "action-plan",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Pending",
                  },
                  "ref": null,
                  "rendered": "Pending",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                    "className": "f-right",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`ActionPlansInfo Component should render metrics 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActionPlansInfo
    metrics={
        Object {
            "completed": 10,
            "created": 10,
            "pending": 10,
          }
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title">
          <p>
                    Action Plans
          </p>
</div>,
        <div
          className="main-content">
          <div
                    className="action-plan">
                    <p>
                              Created
                    </p>
                    <p
                              className="f-right">
                              10
                    </p>
          </div>
          <div
                    className="action-plan border-line">
                    <p>
                              Completed
                    </p>
                    <p
                              className="f-right">
                              10
                    </p>
          </div>
          <div
                    className="action-plan">
                    <p>
                              Pending
                    </p>
                    <p
                              className="f-right">
                              10
                    </p>
          </div>
</div>,
      ],
      "className": "action-plans-info",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            Action Plans
</p>,
          "className": "title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Action Plans",
          },
          "ref": null,
          "rendered": "Action Plans",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="action-plan">
              <p>
                            Created
              </p>
              <p
                            className="f-right">
                            10
              </p>
</div>,
            <div
              className="action-plan border-line">
              <p>
                            Completed
              </p>
              <p
                            className="f-right">
                            10
              </p>
</div>,
            <div
              className="action-plan">
              <p>
                            Pending
              </p>
              <p
                            className="f-right">
                            10
              </p>
</div>,
          ],
          "className": "main-content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Created
</p>,
                <p
                  className="f-right">
                  10
</p>,
              ],
              "className": "action-plan",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Created",
                },
                "ref": null,
                "rendered": "Created",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 10,
                  "className": "f-right",
                },
                "ref": null,
                "rendered": 10,
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Completed
</p>,
                <p
                  className="f-right">
                  10
</p>,
              ],
              "className": "action-plan border-line",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Completed",
                },
                "ref": null,
                "rendered": "Completed",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 10,
                  "className": "f-right",
                },
                "ref": null,
                "rendered": 10,
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  Pending
</p>,
                <p
                  className="f-right">
                  10
</p>,
              ],
              "className": "action-plan",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Pending",
                },
                "ref": null,
                "rendered": "Pending",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 10,
                  "className": "f-right",
                },
                "ref": null,
                "rendered": 10,
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title">
            <p>
                        Action Plans
            </p>
</div>,
          <div
            className="main-content">
            <div
                        className="action-plan">
                        <p>
                                    Created
                        </p>
                        <p
                                    className="f-right">
                                    10
                        </p>
            </div>
            <div
                        className="action-plan border-line">
                        <p>
                                    Completed
                        </p>
                        <p
                                    className="f-right">
                                    10
                        </p>
            </div>
            <div
                        className="action-plan">
                        <p>
                                    Pending
                        </p>
                        <p
                                    className="f-right">
                                    10
                        </p>
            </div>
</div>,
        ],
        "className": "action-plans-info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              Action Plans
</p>,
            "className": "title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Action Plans",
            },
            "ref": null,
            "rendered": "Action Plans",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="action-plan">
                <p>
                                Created
                </p>
                <p
                                className="f-right">
                                10
                </p>
</div>,
              <div
                className="action-plan border-line">
                <p>
                                Completed
                </p>
                <p
                                className="f-right">
                                10
                </p>
</div>,
              <div
                className="action-plan">
                <p>
                                Pending
                </p>
                <p
                                className="f-right">
                                10
                </p>
</div>,
            ],
            "className": "main-content",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Created
</p>,
                  <p
                    className="f-right">
                    10
</p>,
                ],
                "className": "action-plan",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Created",
                  },
                  "ref": null,
                  "rendered": "Created",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 10,
                    "className": "f-right",
                  },
                  "ref": null,
                  "rendered": 10,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Completed
</p>,
                  <p
                    className="f-right">
                    10
</p>,
                ],
                "className": "action-plan border-line",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Completed",
                  },
                  "ref": null,
                  "rendered": "Completed",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 10,
                    "className": "f-right",
                  },
                  "ref": null,
                  "rendered": 10,
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    Pending
</p>,
                  <p
                    className="f-right">
                    10
</p>,
                ],
                "className": "action-plan",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Pending",
                  },
                  "ref": null,
                  "rendered": "Pending",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 10,
                    "className": "f-right",
                  },
                  "ref": null,
                  "rendered": 10,
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
